### PR TITLE
Dev/brih/feature/switch ads to portal context

### LINKED ISSUE #18183: Migrate to Azure wizard hanging on list databases
- Azure Data Studio Version: 1.34.0
- OS Version:  Windows 2012 R2
-  Azure SQL Migration extension has been installed on Azure Data Studio
- Issue
-  I am not able to list the databases of source ( Step2) in migrate to azure SQL wizard- it just hung there , I uninstalled/restart the node/resinstalled the azure studio and tried but the issue remains.
- FYI :- I have approximately 1000 databases on SQL instance 
- Appreciate any help to resolve this.

### PR DESCRIPTION
**Pull request by brian-harris**
This PR fixes #18183

Changes:
* update dashboard to use migration service context instead of saving migration details in global cache
* fixes issue with lost migrations
* fixes slow performance issues with loading migration summary and details lists
* allows user to view all migrations run on service
* update migration service api to: 2022-01-30-preview
for GA release:
* updates to readme 
* update version to v 1.0.0
* remove preview flag from extension

### CODE REVIEW

File: extensions/sql-migration/package.nls.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
 	"start-migration-command": "Migrate to Azure SQL",
 	"send-feedback-command": "Feedback",
 	"new-support-request-command": "New support request",
+	"refresh-migrations-command": "Refresh",
```
Comment (aasimkhan30) [inline extensions/sql-migration/package.nls.json:11]: Please remove this resource as well if you do not plan on using it.

File: extensions/sql-migration/package.json [hunk 2]
```diff
@@ -104,6 +111,10 @@
 					"command": "sqlmigration.sendfeedback",
 					"when": "false"
 				},
+				{
+					"command": "sqlmigration.refreshmigrations",
```
[the remark this answers is not in the document]
Comment (brian-harris) [reply extensions/sql-migration/package.json:115]: This is needed as it adds a different refresh button that works, but it's a duplicate.  I was not able to hook up to the default refresh button you are hiding in a different PR.

File: extensions/sql-migration/package.nls.json (hunk 1 above)
Comment (brian-harris) [reply extensions/sql-migration/package.nls.json:11]: This is used. a Refresh button/command I am adding to the commands list for the dashboard

Comment (brian-harris) [reply extensions/sql-migration/package.nls.json:11]: This is used. a Refresh button/command I am adding to the commands list for the dashboard

**Review by raymondtruong — CHANGES_REQUESTED** [review]

File: extensions/sql-migration/src/dialog/selectMigrationService/selectMigrationServiceDialog.ts [hunk 3]
```diff
@@ -0,0 +1,524 @@
...
+	'font-weight': '600',
+};
+const DROPDOWN_CSS = {
+	'margin': '-1em 0 0 0',
+};
+const TENANT_DROPDOWN_CSS = {
+	'margin': '1em 0 0 0',
+};
+
+export class SelectMigrationServiceDialog {
```
[the remark this answers is not in the document]
Comment (brian-harris) [reply extensions/sql-migration/src/dialog/selectMigrationService/selectMigrationServiceDialog.ts:38]: this is a bug in the dropdown control

File: extensions/sql-migration/src/api/azure.ts [hunk 4]
```diff
@@ -142,17 +142,36 @@ export async function getBlobs(account: azdata.Account, subscription: Subscripti
...
+export async function getSqlMigrationService(account: azdata.Account, subscription: Subscription, resourceGroupName: string, regionName: string, sqlMigrationServiceName: string): Promise<SqlMigrationService> {
+	const sqlMigrationServiceId = `/subscriptions/${subscription.id}/resourceGroups/${resourceGroupName}/providers/Microsoft.DataMigration/sqlMigrationServices/${sqlMigrationServiceName}`;
+	return await getSqlMigrationServiceById(account, subscription, sqlMigrationServiceId);
+}
+
+export async function getSqlMigrationServiceById(account: azdata.Account, subscription: Subscription, sqlMigrationServiceId: string): Promise<SqlMigrationService> {
 	const api = await getAzureCoreAPI();
-	const path = encodeURI(`/subscriptions/${subscription.id}/resourceGroups/${resourceGroupName}/providers/Microsoft.DataMigration/sqlMigrationServices/${sqlMigrationServiceName}?api-version=2020-09-01-preview`);
-	const response = await api.makeAzureRestRequest(account, subscription, path, azurecore.HttpRequestMethod.GET, undefined, true, undefined, getSessionIdHeader(sessionId));
+	const path = encodeURI(`${sqlMigrationServiceId}?api-version=2022-01-30-preview`);
```
[the remark this answers is not in the document]
Comment (brian-harris) [reply extensions/sql-migration/src/api/azure.ts:152]: what will the new api offer/change/break

File: extensions/sql-migration/src/constants/strings.ts [hunk 5]
```diff
@@ -505,9 +507,9 @@ export const PRE_REQ_1 = localize('sql.migration.pre.req.1', "Azure account deta
 export const PRE_REQ_2 = localize('sql.migration.pre.req.2', "Azure SQL Managed Instance or SQL Server on Azure Virtual Machine");
 export const PRE_REQ_3 = localize('sql.migration.pre.req.3', "Backup location details");
 export const MIGRATION_IN_PROGRESS = localize('sql.migration.migration.in.progress', "Database migrations in progress");
-export const MIGRATION_FAILED = localize('sql.migration.failed', "Migrations failed");
-export const MIGRATION_COMPLETED = localize('sql.migration.migration.completed', "Migrations completed");
-export const MIGRATION_CUTOVER_CARD = localize('sql.migration.cutover.card', "Completing cutover");
+export const MIGRATION_FAILED = localize('sql.migration.failed', "Database migrations failed");
+export const MIGRATION_COMPLETED = localize('sql.migration.migration.completed', "Database migrations completed");
+export const MIGRATION_CUTOVER_CARD = localize('sql.migration.cutover.card', "Database migrations completing cutover");
 export const MIGRATION_NOT_STARTED = localize('sql.migration.migration.not.started', "Migrations not started");
```
[the remark this answers is not in the document]
Comment (brian-harris) [reply extensions/sql-migration/src/constants/strings.ts:531]: it's not used or related to my change

File: extensions/sql-migration/src/dialog/selectMigrationService/selectMigrationServiceDialog.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (raymondtruong) [reply extensions/sql-migration/src/dialog/selectMigrationService/selectMigrationServiceDialog.ts:38]: Got it - but for what it's worth I saw it here significantly more often than elsewhere in the wizard, even for the page where the user selects their DMS service (which presumably does the exact same thing this dialog does)

File: extensions/sql-migration/src/api/azure.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (raymondtruong) [reply extensions/sql-migration/src/api/azure.ts:152]: As far as I know, nothing relevant to us, but figured we'd just bump it to the latest version. I believe we denoted this `2022-03-30-preview` as our "official" GA release version, but it doesn't really matter.

**Review by Rheeser — APPROVED** [review]

**Review by raymondtruong — APPROVED** [review]

**Pull request merged**